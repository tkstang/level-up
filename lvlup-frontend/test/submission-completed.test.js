import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import SubmissionCompleted from '../src/components/Student/submissions/submission-completed';
import lvlupApp from '../src/reducers/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

test('SubmissionCompleted should render', () => {
  const store = mockStore({ lvlupApp });
  const sub = shallow(
    <SubmissionCompleted submission={{ submission_message: 'Good job' }} selectedChallenge={{ name: 'Feed the Hamid' }} store={store} />,
  );
  expect(shallowToJson(sub)).toMatchSnapshot();
});
