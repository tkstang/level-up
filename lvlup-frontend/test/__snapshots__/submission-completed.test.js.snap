exports[`test SubmissionCompleted should render 1`] = `
<div
  className="lvl-table">
  <Table
    as="table"
    celled={true}
    selectable={true}>
    <TableHeader
      as="thead">
      <TableRow
        as="tr"
        cellAs="td">
        <TableHeaderCell
          as="th"
          className="completed-thead"
          textAlign="center">
          Challenge Submission Successful!
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody">
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Name
          </Label>
          Feed the Hamid
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td">
        <TableCell
          as="td">
          <Label
            ribbon={true}>
            Message
          </Label>
          Good job
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Link
    replace={false}
    to="/student/challenges">
    <Button
      as="button"
      basic={true}
      color="orange">
      Back to Challenges
    </Button>
  </Link>
</div>
`;
